In [1]:
import pandas as pd

In [2]:
df_vaers_data = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2021VAERSDATA.csv')
df_vaers_symptoms = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2021VAERSSYMPTOMS.csv')
df_vaers_vax = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2021VAERSVAX.csv')
df_vaers_data_2020 = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2020VAERSDATA.csv')
df_vaers_symptoms_2020 = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2020VAERSSYMPTOMS.csv')
df_vaers_vax_2020 = pd.read_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\2020VAERSVAX.csv')
df_vaers_data = pd.concat([df_vaers_data,df_vaers_data_2020],axis=0)
df_vaers_symptoms = pd.concat([df_vaers_symptoms,df_vaers_symptoms_2020],axis=0)
df_vaers_vax = pd.concat([df_vaers_vax,df_vaers_vax_2020],axis=0)

C:\Users\NIKHIL\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (7,12,15,23) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\NIKHIL\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (7,12,15,23,31) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
df_vaers_data

VAERS_ID    RECVDATE STATE  AGE_YRS  CAGE_YR  CAGE_MO SEX RPT_DATE  \
0        916600    1/1/2021    TX     33.0     33.0      NaN   F      NaN   
1        916601    1/1/2021    CA     73.0     73.0      NaN   F      NaN   
2        916602    1/1/2021    WA     23.0     23.0      NaN   F      NaN   
3        916603    1/1/2021    WA     58.0     58.0      NaN   F      NaN   
4        916604    1/1/2021    TX     47.0     47.0      NaN   F      NaN   
...         ...         ...   ...      ...      ...      ...  ..      ...   
50140    918441  12/31/2020    NE     35.0     35.0      NaN   F      NaN   
50141    918442  12/31/2020    AR      NaN     34.0      NaN   F      NaN   
50142    918503  12/31/2020    RI     50.0     50.0      NaN   F      NaN   
50143    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
50144    918561  12/31/2020    WA     95.0     95.0      NaN   F      NaN   

                                            SYMPTOM_TEXT DIED  ...  \
0      Right side of epiglottis swelled up and hinder...  NaN  ...   
1      Approximately 30 min post vaccination administ...  NaN  ...   
2      About 15 minutes after receiving the vaccine, ...  NaN  ...   
3      extreme fatigue, dizziness,. could not lift my...  NaN  ...   
4      Injection site swelling, redness, warm to the ...  NaN  ...   
...                                                  ...  ...  ...   
50140    Injection site is hot, tender to touch, swollen  NaN  ...   
50141  Injection site - swollen arm  Muscle aches Joi...  NaN  ...   
50142  Nausea, tightness in chest from nausea  BP  10...  NaN  ...   
50143  10 days after vaccines, developed hives & indu...  NaN  ...   
50144  Report of awakening with difficulty concentrat...  NaN  ...   

                                                 CUR_ILL  \
0                                                   None   
1      Patient residing at nursing facility. See pati...   
2                                                   None   
3                                       kidney infection   
4                                                     Na   
...                                                  ...   
50140                                               None   
50141                                                NaN   
50142                                               None   
50143                                                NaN   
50144                     Fatigue, elevated hypertension   

                                                 HISTORY  \
0                                                   None   
1      Patient residing at nursing facility. See pati...   
2                                                   None   
3      diverticulitis, mitral valve prolapse, osteoar...   
4                                                    NaN   
...                                                  ...   
50140                                               None   
50141                                                NaN   
50142                                             Asthma   
50143                                                NaN   
50144  Hypertension, DM type 2, hyperlipidemia, hypot...   

                                               PRIOR_VAX SPLTTYPE  FORM_VERS  \
0                                                    NaN      NaN          2   
1                                                    NaN      NaN          2   
2                                                    NaN      NaN          2   
3      got measles from measel shot, mums from mumps ...      NaN          2   
4                                                    NaN      NaN          2   
...                                                  ...      ...        ...   
50140                                                NaN      NaN          2   
50141                                                NaN      NaN          2   
50142                                                NaN      NaN          2   

In [4]:
df_vaers_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 588107 entries, 0 to 50144
Data columns (total 35 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   VAERS_ID      588107 non-null  int64  
 1   RECVDATE      588107 non-null  object 
 2   STATE         517073 non-null  object 
 3   AGE_YRS       522691 non-null  float64
 4   CAGE_YR       477382 non-null  float64
 5   CAGE_MO       4509 non-null    float64
 6   SEX           588107 non-null  object 
 7   RPT_DATE      1354 non-null    object 
 8   SYMPTOM_TEXT  587977 non-null  object 
 9   DIED          6938 non-null    object 
 10  DATEDIED      6190 non-null    object 
 11  L_THREAT      8766 non-null    object 
 12  ER_VISIT      226 non-null     object 
 13  HOSPITAL      32722 non-null   object 
 14  HOSPDAYS      22026 non-null   float64
 15  X_STAY        327 non-null     object 
 16  DISABLE       9175 non-null    object 
 17  RECOVD        537181 non-null  object 
 18  VAX_D

In [5]:
df_vaers_symptoms

VAERS_ID                   SYMPTOM1  SYMPTOMVERSION1  \
0        916600                  Dysphagia             23.1   
1        916601                    Anxiety             23.1   
2        916602           Chest discomfort             23.1   
3        916603                  Dizziness             23.1   
4        916604    Injection site erythema             23.1   
...         ...                        ...              ...   
61495    918441        Injection site pain             23.1   
61496    918442                 Arthralgia             23.1   
61497    918503           Chest discomfort             23.1   
61498    918512  Injection site induration             24.0   
61499    918561       Blood glucose normal             23.1   

                      SYMPTOM2  SYMPTOMVERSION2                  SYMPTOM3  \
0                 Epiglottitis             23.1                       NaN   
1                     Dyspnoea             23.1                       NaN   
2                    Dysphagia             23.1         Pain in extremity   
3                      Fatigue             23.1        Mobility decreased   
4      Injection site pruritus             23.1   Injection site swelling   
...                        ...              ...                       ...   
61495  Injection site swelling             23.1     Injection site warmth   
61496                Dysgeusia             23.1       Injection site pain   
61497                   Nausea             23.1                       NaN   
61498                Urticaria             24.0                       NaN   
61499        Confusional state             23.1  Disturbance in attention   

       SYMPTOMVERSION3                  SYMPTOM4  SYMPTOMVERSION4  \
0                  NaN                       NaN              NaN   
1                  NaN                       NaN              NaN   
2                 23.1         Visual impairment             23.1   
3                 23.1                       NaN              NaN   
4                 23.1     Injection site warmth             23.1   
...                ...                       ...              ...   
61495             23.1                       NaN              NaN   
61496             23.1   Injection site swelling             23.1   
61497              NaN                       NaN              NaN   
61498              NaN                       NaN              NaN   
61499             23.1  Electrocardiogram normal             23.1   

           SYMPTOM5  SYMPTOMVERSION5  
0               NaN              NaN  
1               NaN              NaN  
2               NaN              NaN  
3               NaN              NaN  
4               NaN              NaN  
...             ...              ...  
61495           NaN              NaN  
61496  Paraesthesia             23.1  
61497           NaN              NaN  
61498           NaN              NaN  
61499          Fear             23.1  

[778655 rows x 11 columns]

In [6]:
df_vaers_symptoms.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 778655 entries, 0 to 61499
Data columns (total 11 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   VAERS_ID         778655 non-null  int64  
 1   SYMPTOM1         778655 non-null  object 
 2   SYMPTOMVERSION1  778655 non-null  float64
 3   SYMPTOM2         611032 non-null  object 
 4   SYMPTOMVERSION2  611032 non-null  float64
 5   SYMPTOM3         469883 non-null  object 
 6   SYMPTOMVERSION3  469883 non-null  float64
 7   SYMPTOM4         353940 non-null  object 
 8   SYMPTOMVERSION4  353940 non-null  float64
 9   SYMPTOM5         260847 non-null  object 
 10  SYMPTOMVERSION5  260847 non-null  float64
dtypes: float64(5), int64(1), object(5)
memory usage: 71.3+ MB


In [7]:
df_vaers_vax

VAERS_ID VAX_TYPE          VAX_MANU  VAX_LOT VAX_DOSE_SERIES VAX_ROUTE  \
0        916600  COVID19           MODERNA  037K20A               1        IM   
1        916601  COVID19           MODERNA  025L20A               1        IM   
2        916602  COVID19   PFIZER\BIONTECH   EL1284               1        IM   
3        916603  COVID19           MODERNA  unknown             UNK       NaN   
4        916604  COVID19           MODERNA      NaN               1        IM   
...         ...      ...               ...      ...             ...       ...   
60143    918503  COVID19           MODERNA  039K20A               1        IM   
60144    918512      MMR  MERCK & CO. INC.  S029457               1        SC   
60145    918512    PNC13      PFIZER\WYETH   DJ7723               4        IM   
60146    918512   VARCEL  MERCK & CO. INC.  T020652               1        SC   
60147    918561  COVID19   PFIZER\BIONTECH   EJ1685               1        IM   

      VAX_SITE                             VAX_NAME  
0           LA          COVID19 (COVID19 (MODERNA))  
1           RA          COVID19 (COVID19 (MODERNA))  
2           LA  COVID19 (COVID19 (PFIZER-BIONTECH))  
3          NaN          COVID19 (COVID19 (MODERNA))  
4           LA          COVID19 (COVID19 (MODERNA))  
...        ...                                  ...  
60143       RA          COVID19 (COVID19 (MODERNA))  
60144       LL   MEASLES + MUMPS + RUBELLA (MMR II)  
60145       LL                   PNEUMO (PREVNAR13)  
60146       RL                  VARICELLA (VARIVAX)  
60147       LA  COVID19 (COVID19 (PFIZER-BIONTECH))  

[620484 rows x 8 columns]

In [8]:
df_vaers_vax.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 620484 entries, 0 to 60147
Data columns (total 8 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   VAERS_ID         620484 non-null  int64 
 1   VAX_TYPE         620484 non-null  object
 2   VAX_MANU         620484 non-null  object
 3   VAX_LOT          425027 non-null  object
 4   VAX_DOSE_SERIES  616622 non-null  object
 5   VAX_ROUTE        484392 non-null  object
 6   VAX_SITE         456292 non-null  object
 7   VAX_NAME         620484 non-null  object
dtypes: int64(1), object(7)
memory usage: 42.6+ MB


In [9]:
df_merge_vaers = pd.merge(left=df_vaers_data,right=df_vaers_symptoms,on='VAERS_ID')
df_merge_vaers = pd.merge(left=df_merge_vaers,right=df_vaers_vax,on='VAERS_ID')
df_merge_vaers

VAERS_ID    RECVDATE STATE  AGE_YRS  CAGE_YR  CAGE_MO SEX RPT_DATE  \
0         916600    1/1/2021    TX     33.0     33.0      NaN   F      NaN   
1         916601    1/1/2021    CA     73.0     73.0      NaN   F      NaN   
2         916602    1/1/2021    WA     23.0     23.0      NaN   F      NaN   
3         916603    1/1/2021    WA     58.0     58.0      NaN   F      NaN   
4         916604    1/1/2021    TX     47.0     47.0      NaN   F      NaN   
...          ...         ...   ...      ...      ...      ...  ..      ...   
825185    918503  12/31/2020    RI     50.0     50.0      NaN   F      NaN   
825186    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825187    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825188    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825189    918561  12/31/2020    WA     95.0     95.0      NaN   F      NaN   

                                             SYMPTOM_TEXT DIED  ...  \
0       Right side of epiglottis swelled up and hinder...  NaN  ...   
1       Approximately 30 min post vaccination administ...  NaN  ...   
2       About 15 minutes after receiving the vaccine, ...  NaN  ...   
3       extreme fatigue, dizziness,. could not lift my...  NaN  ...   
4       Injection site swelling, redness, warm to the ...  NaN  ...   
...                                                   ...  ...  ...   
825185  Nausea, tightness in chest from nausea  BP  10...  NaN  ...   
825186  10 days after vaccines, developed hives & indu...  NaN  ...   
825187  10 days after vaccines, developed hives & indu...  NaN  ...   
825188  10 days after vaccines, developed hives & indu...  NaN  ...   
825189  Report of awakening with difficulty concentrat...  NaN  ...   

       SYMPTOMVERSION4 SYMPTOM5 SYMPTOMVERSION5 VAX_TYPE          VAX_MANU  \
0                  NaN      NaN             NaN  COVID19           MODERNA   
1                  NaN      NaN             NaN  COVID19           MODERNA   
2                 23.1      NaN             NaN  COVID19   PFIZER\BIONTECH   
3                  NaN      NaN             NaN  COVID19           MODERNA   
4                 23.1      NaN             NaN  COVID19           MODERNA   
...                ...      ...             ...      ...               ...   
825185             NaN      NaN             NaN  COVID19           MODERNA   
825186             NaN      NaN             NaN      MMR  MERCK & CO. INC.   
825187             NaN      NaN             NaN    PNC13      PFIZER\WYETH   
825188             NaN      NaN             NaN   VARCEL  MERCK & CO. INC.   
825189            23.1     Fear            23.1  COVID19   PFIZER\BIONTECH   

        VAX_LOT VAX_DOSE_SERIES VAX_ROUTE VAX_SITE  \
0       037K20A               1        IM       LA   
1       025L20A               1        IM       RA   
2        EL1284               1        IM       LA   
3       unknown             UNK       NaN      NaN   
4           NaN               1        IM       LA   
...         ...             ...       ...      ...   
825185  039K20A               1        IM       RA   
825186  S029457               1        SC       LL   
825187   DJ7723               4        IM       LL   
825188  T020652               1        SC       RL   
825189   EJ1685               1        IM       LA   

                                   VAX_NAME  
0               COVID19 (COVID19 (MODERNA))  
1               COVID19 (COVID19 (MODERNA))  
2       COVID19 (COVID19 (PFIZER-BIONTECH))  
3               COVID19 (COVID19 (MODERNA))  
4               COVID19 (COVID19 (MODERNA))  
...                                     ...  
825185          COVID19 (COVID19 (MODERNA))  
825186   MEASLES + MUMPS + RUBELLA (MMR II)  
825187                   PNEUMO (PREVNAR13)  
825188                  VARICELLA (VARIVAX)  
825189  COVID19 (COVID19 (PFIZER-BIONTECH))  

[825190 rows x 52 columns]

In [10]:
df_merge_vaers.shape

(825190, 52)

In [11]:
df_merge_vaers.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 825190 entries, 0 to 825189
Data columns (total 52 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   VAERS_ID         825190 non-null  int64  
 1   RECVDATE         825190 non-null  object 
 2   STATE            737801 non-null  object 
 3   AGE_YRS          747933 non-null  float64
 4   CAGE_YR          683795 non-null  float64
 5   CAGE_MO          10420 non-null   float64
 6   SEX              825190 non-null  object 
 7   RPT_DATE         2074 non-null    object 
 8   SYMPTOM_TEXT     825049 non-null  object 
 9   DIED             12260 non-null   object 
 10  DATEDIED         11306 non-null   object 
 11  L_THREAT         20507 non-null   object 
 12  ER_VISIT         415 non-null     object 
 13  HOSPITAL         72673 non-null   object 
 14  HOSPDAYS         52518 non-null   float64
 15  X_STAY           723 non-null     object 
 16  DISABLE          20206 non-null   obje

In [12]:
df_merge_vaers.head()

VAERS_ID  RECVDATE STATE  AGE_YRS  CAGE_YR  CAGE_MO SEX RPT_DATE  \
0    916600  1/1/2021    TX     33.0     33.0      NaN   F      NaN   
1    916601  1/1/2021    CA     73.0     73.0      NaN   F      NaN   
2    916602  1/1/2021    WA     23.0     23.0      NaN   F      NaN   
3    916603  1/1/2021    WA     58.0     58.0      NaN   F      NaN   
4    916604  1/1/2021    TX     47.0     47.0      NaN   F      NaN   

                                        SYMPTOM_TEXT DIED  ...  \
0  Right side of epiglottis swelled up and hinder...  NaN  ...   
1  Approximately 30 min post vaccination administ...  NaN  ...   
2  About 15 minutes after receiving the vaccine, ...  NaN  ...   
3  extreme fatigue, dizziness,. could not lift my...  NaN  ...   
4  Injection site swelling, redness, warm to the ...  NaN  ...   

  SYMPTOMVERSION4 SYMPTOM5 SYMPTOMVERSION5 VAX_TYPE         VAX_MANU  VAX_LOT  \
0             NaN      NaN             NaN  COVID19          MODERNA  037K20A   
1             NaN      NaN             NaN  COVID19          MODERNA  025L20A   
2            23.1      NaN             NaN  COVID19  PFIZER\BIONTECH   EL1284   
3             NaN      NaN             NaN  COVID19          MODERNA  unknown   
4            23.1      NaN             NaN  COVID19          MODERNA      NaN   

  VAX_DOSE_SERIES VAX_ROUTE VAX_SITE                             VAX_NAME  
0               1        IM       LA          COVID19 (COVID19 (MODERNA))  
1               1        IM       RA          COVID19 (COVID19 (MODERNA))  
2               1        IM       LA  COVID19 (COVID19 (PFIZER-BIONTECH))  
3             UNK       NaN      NaN          COVID19 (COVID19 (MODERNA))  
4               1        IM       LA          COVID19 (COVID19 (MODERNA))  

[5 rows x 52 columns]

In [13]:
df_merge_vaers.tail()

VAERS_ID    RECVDATE STATE  AGE_YRS  CAGE_YR  CAGE_MO SEX RPT_DATE  \
825185    918503  12/31/2020    RI     50.0     50.0      NaN   F      NaN   
825186    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825187    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825188    918512  12/31/2020    OR      1.0      1.0      0.0   M      NaN   
825189    918561  12/31/2020    WA     95.0     95.0      NaN   F      NaN   

                                             SYMPTOM_TEXT DIED  ...  \
825185  Nausea, tightness in chest from nausea  BP  10...  NaN  ...   
825186  10 days after vaccines, developed hives & indu...  NaN  ...   
825187  10 days after vaccines, developed hives & indu...  NaN  ...   
825188  10 days after vaccines, developed hives & indu...  NaN  ...   
825189  Report of awakening with difficulty concentrat...  NaN  ...   

       SYMPTOMVERSION4 SYMPTOM5 SYMPTOMVERSION5 VAX_TYPE          VAX_MANU  \
825185             NaN      NaN             NaN  COVID19           MODERNA   
825186             NaN      NaN             NaN      MMR  MERCK & CO. INC.   
825187             NaN      NaN             NaN    PNC13      PFIZER\WYETH   
825188             NaN      NaN             NaN   VARCEL  MERCK & CO. INC.   
825189            23.1     Fear            23.1  COVID19   PFIZER\BIONTECH   

        VAX_LOT VAX_DOSE_SERIES VAX_ROUTE VAX_SITE  \
825185  039K20A               1        IM       RA   
825186  S029457               1        SC       LL   
825187   DJ7723               4        IM       LL   
825188  T020652               1        SC       RL   
825189   EJ1685               1        IM       LA   

                                   VAX_NAME  
825185          COVID19 (COVID19 (MODERNA))  
825186   MEASLES + MUMPS + RUBELLA (MMR II)  
825187                   PNEUMO (PREVNAR13)  
825188                  VARICELLA (VARIVAX)  
825189  COVID19 (COVID19 (PFIZER-BIONTECH))  

[5 rows x 52 columns]

In [14]:
df_merge_vaers.drop(['RPT_DATE', 'ER_VISIT','FORM_VERS','SPLTTYPE','CAGE_YR','CAGE_MO','V_FUNDBY','VAX_SITE','VAX_DOSE_SERIES','VAX_LOT','V_ADMINBY'],axis='columns',inplace=True)
df_merge_vaers.shape

(825190, 41)

In [15]:
df_merge_vaers.VAX_TYPE.unique()

array(['COVID19', 'FLUC4', 'UNK', 'FLU4', 'PPV', 'FLUA3', 'VARZOS',
       'DTAPHEPBIP', 'HIBV', 'RV1', 'HPV9', 'DTAP', 'MMRV', 'TDAP',
       'FLUR4', 'PNC13', 'YF', 'FLUX', 'HEP', 'FLUA4', 'FLUC3', 'HEPA',
       'HPV4', 'MMR', 'MENB', 'FLUN4', 'DTAPIPV', 'ADEN_4_7', 'ANTH',
       'VARCEL', 'RV5', 'CHOL', 'DT', 'TTOX', 'FLU3', 'MNQ', 'DTAPIPVHIB',
       'EBZR', 'IPV', 'TYP', 'DF', 'HPVX', 'TD', 'FLUX(H1N1)', 'RVX',
       'HEPAB', 'MENHIB', 'DTP', 'JEV1', 'FLU(H1N1)', 'RAB', 'PNC', 'MEN',
       'MNQHIB', 'OPV', 'SMALL', 'FLUN3', '6VAX-F', 'PNC10', 'FLUR3',
       'HPV2', 'BCG', 'JEVX', 'DTOX', 'HBHEPB', 'RUB'], dtype=object)

In [16]:
len(df_merge_vaers[df_merge_vaers['VAX_TYPE'] != 'COVID19'])

77770

In [17]:
df_merge_vaers.shape[0] - len(df_merge_vaers[df_merge_vaers['VAX_TYPE'] != 'COVID19'])

747420

In [18]:
df_merge_vaers.drop(df_merge_vaers[df_merge_vaers.VAX_TYPE != 'COVID19'].index, inplace=True )

In [19]:
df_merge_vaers.shape[0]

747420

In [20]:
df_merge_vaers.VAX_TYPE.unique()

array(['COVID19'], dtype=object)

In [21]:
df_merge_vaers.head()

VAERS_ID  RECVDATE STATE  AGE_YRS SEX  \
0    916600  1/1/2021    TX     33.0   F   
1    916601  1/1/2021    CA     73.0   F   
2    916602  1/1/2021    WA     23.0   F   
3    916603  1/1/2021    WA     58.0   F   
4    916604  1/1/2021    TX     47.0   F   

                                        SYMPTOM_TEXT DIED DATEDIED L_THREAT  \
0  Right side of epiglottis swelled up and hinder...  NaN      NaN      NaN   
1  Approximately 30 min post vaccination administ...  NaN      NaN      NaN   
2  About 15 minutes after receiving the vaccine, ...  NaN      NaN      NaN   
3  extreme fatigue, dizziness,. could not lift my...  NaN      NaN      NaN   
4  Injection site swelling, redness, warm to the ...  NaN      NaN      NaN   

  HOSPITAL  ...                 SYMPTOM3 SYMPTOMVERSION3  \
0      NaN  ...                      NaN             NaN   
1      NaN  ...                      NaN             NaN   
2      NaN  ...        Pain in extremity            23.1   
3      NaN  ...       Mobility decreased            23.1   
4      NaN  ...  Injection site swelling            23.1   

                SYMPTOM4 SYMPTOMVERSION4 SYMPTOM5 SYMPTOMVERSION5  VAX_TYPE  \
0                    NaN             NaN      NaN             NaN   COVID19   
1                    NaN             NaN      NaN             NaN   COVID19   
2      Visual impairment            23.1      NaN             NaN   COVID19   
3                    NaN             NaN      NaN             NaN   COVID19   
4  Injection site warmth            23.1      NaN             NaN   COVID19   

          VAX_MANU VAX_ROUTE                             VAX_NAME  
0          MODERNA        IM          COVID19 (COVID19 (MODERNA))  
1          MODERNA        IM          COVID19 (COVID19 (MODERNA))  
2  PFIZER\BIONTECH        IM  COVID19 (COVID19 (PFIZER-BIONTECH))  
3          MODERNA       NaN          COVID19 (COVID19 (MODERNA))  
4          MODERNA        IM          COVID19 (COVID19 (MODERNA))  

[5 rows x 41 columns]

In [22]:
df_merge_vaers["RECVDATE"] = pd.to_datetime(df_merge_vaers["RECVDATE"])
df_merge_vaers = df_merge_vaers.sort_values(by='RECVDATE')
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ     56.0   F   
808126    902557 2020-12-15    AK     32.0   F   
808124    902524 2020-12-15    NE     59.0   F   
808123    902524 2020-12-15    NE     59.0   F   
808113    902518 2020-12-15    IN     58.0   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...  NaN      NaN   
808126  1750- IM injection ot R Deltoid.  She was sitt...  NaN      NaN   
808124  I am a immunization nurse at this location.  I...  NaN      NaN   
808123  I am a immunization nurse at this location.  I...  NaN      NaN   
808113  Tingling of upper lip and cheeks, warmth in fa...  NaN      NaN   

       L_THREAT HOSPITAL  ...          SYMPTOM3 SYMPTOMVERSION3  \
807983      NaN      NaN  ...               NaN             NaN   
808126      NaN      NaN  ...     Tunnel vision            23.1   
808124      NaN      NaN  ...       Nervousness            23.1   
808123      NaN      NaN  ...  Feeling abnormal            23.1   
808113      NaN      NaN  ...      Paraesthesia            23.1   

                    SYMPTOM4 SYMPTOMVERSION4      SYMPTOM5 SYMPTOMVERSION5  \
807983                   NaN             NaN           NaN             NaN   
808126                   NaN             NaN           NaN             NaN   
808124                   NaN             NaN           NaN             NaN   
808123  Heart rate increased            23.1  Hypertension            23.1   
808113     Paraesthesia oral            23.1           NaN             NaN   

        VAX_TYPE         VAX_MANU VAX_ROUTE  \
807983   COVID19  PFIZER\BIONTECH        IM   
808126   COVID19  PFIZER\BIONTECH        IM   
808124   COVID19  PFIZER\BIONTECH        IM   
808123   COVID19  PFIZER\BIONTECH        IM   
808113   COVID19  PFIZER\BIONTECH        IM   

                                   VAX_NAME  
807983  COVID19 (COVID19 (PFIZER-BIONTECH))  
808126  COVID19 (COVID19 (PFIZER-BIONTECH))  
808124  COVID19 (COVID19 (PFIZER-BIONTECH))  
808123  COVID19 (COVID19 (PFIZER-BIONTECH))  
808113  COVID19 (COVID19 (PFIZER-BIONTECH))  

[5 rows x 41 columns]

In [23]:
df_merge_vaers.DIED.unique()

array([nan, 'Y'], dtype=object)

In [24]:
df_merge_vaers['DIED'] = df_merge_vaers['DIED'].fillna('N')
df_merge_vaers['DATEDIED'] = df_merge_vaers['DATEDIED'].fillna('NA')
df_merge_vaers['L_THREAT'] = df_merge_vaers['L_THREAT'].fillna('N')
df_merge_vaers['HOSPITAL'] = df_merge_vaers['HOSPITAL'].fillna('N')

In [25]:
df_merge_vaers.DIED.unique()

array(['N', 'Y'], dtype=object)

In [26]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ     56.0   F   
808126    902557 2020-12-15    AK     32.0   F   
808124    902524 2020-12-15    NE     59.0   F   
808123    902524 2020-12-15    NE     59.0   F   
808113    902518 2020-12-15    IN     58.0   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  ...          SYMPTOM3 SYMPTOMVERSION3  \
807983        N        N  ...               NaN             NaN   
808126        N        N  ...     Tunnel vision            23.1   
808124        N        N  ...       Nervousness            23.1   
808123        N        N  ...  Feeling abnormal            23.1   
808113        N        N  ...      Paraesthesia            23.1   

                    SYMPTOM4 SYMPTOMVERSION4      SYMPTOM5 SYMPTOMVERSION5  \
807983                   NaN             NaN           NaN             NaN   
808126                   NaN             NaN           NaN             NaN   
808124                   NaN             NaN           NaN             NaN   
808123  Heart rate increased            23.1  Hypertension            23.1   
808113     Paraesthesia oral            23.1           NaN             NaN   

        VAX_TYPE         VAX_MANU VAX_ROUTE  \
807983   COVID19  PFIZER\BIONTECH        IM   
808126   COVID19  PFIZER\BIONTECH        IM   
808124   COVID19  PFIZER\BIONTECH        IM   
808123   COVID19  PFIZER\BIONTECH        IM   
808113   COVID19  PFIZER\BIONTECH        IM   

                                   VAX_NAME  
807983  COVID19 (COVID19 (PFIZER-BIONTECH))  
808126  COVID19 (COVID19 (PFIZER-BIONTECH))  
808124  COVID19 (COVID19 (PFIZER-BIONTECH))  
808123  COVID19 (COVID19 (PFIZER-BIONTECH))  
808113  COVID19 (COVID19 (PFIZER-BIONTECH))  

[5 rows x 41 columns]

In [27]:
df_merge_vaers.SEX.unique()

array(['F', 'M', 'U'], dtype=object)

In [28]:
len(df_merge_vaers[df_merge_vaers['SEX'] == 'U'])

19214

In [29]:
pd.set_option("display.max_columns",None)
pd.set_option("display.float_format",lambda x:'%.2f'%x)
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N       NaN    NaN     NaN      Y  12/15/2020   
808126        N        N       NaN    NaN     NaN      U  12/15/2020   
808124        N        N       NaN    NaN     NaN      Y  12/15/2020   
808123        N        N       NaN    NaN     NaN      Y  12/15/2020   
808113        N        N       NaN    NaN     NaN      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [30]:
df_merge_vaers.drop_duplicates(subset=None,keep='first',inplace=True)

In [31]:
df_merge_vaers.shape

(721435, 41)

In [32]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N       NaN    NaN     NaN      Y  12/15/2020   
808126        N        N       NaN    NaN     NaN      U  12/15/2020   
808124        N        N       NaN    NaN     NaN      Y  12/15/2020   
808123        N        N       NaN    NaN     NaN      Y  12/15/2020   
808113        N        N       NaN    NaN     NaN      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [33]:
df_merge_vaers.HOSPDAYS.unique()

array([       nan, 1.0000e+00, 2.0000e+00, 3.0000e+00, 4.0000e+00,
       6.0000e+00, 9.0000e+00, 7.0000e+00, 5.0000e+00, 1.1000e+01,
       1.0000e+01, 8.0000e+00, 1.4000e+01, 1.5000e+01, 2.2000e+01,
       1.6000e+01, 1.3000e+01, 1.7000e+01, 1.2000e+01, 2.0000e+01,
       2.1000e+01, 2.6000e+01, 2.4000e+01, 3.6000e+01, 1.9000e+01,
       2.5000e+01, 1.8000e+01, 2.7000e+01, 2.9000e+01, 3.9000e+01,
       2.8000e+01, 3.3000e+01, 3.8000e+01, 3.7000e+01, 3.0000e+01,
       4.3000e+01, 4.2000e+01, 4.0000e+01, 3.4000e+01, 3.2000e+01,
       2.3000e+01, 3.1000e+01, 9.0000e+01, 5.5000e+01, 5.0000e+01,
       3.5000e+01, 4.5000e+01, 9.9900e+02, 5.1000e+01, 4.4000e+01,
       4.1000e+01, 4.7000e+01, 1.0400e+02, 6.5000e+01, 6.8000e+01,
       5.9000e+01, 5.4000e+01, 6.0000e+01, 8.0000e+01, 9.9999e+04,
       7.8000e+01, 6.3000e+01, 8.2000e+01, 9.5000e+01, 9.1000e+01,
       5.7000e+01, 9.7000e+01, 1.2000e+02, 7.4000e+01, 9.2000e+01,
       7.5000e+01, 4.9000e+01, 5.2000e+01, 1.1700e+02, 5.6000e

In [34]:
df_merge_vaers['HOSPDAYS'] = df_merge_vaers.HOSPDAYS.fillna(0)

In [35]:
df_merge_vaers.HOSPDAYS.unique()

array([0.0000e+00, 1.0000e+00, 2.0000e+00, 3.0000e+00, 4.0000e+00,
       6.0000e+00, 9.0000e+00, 7.0000e+00, 5.0000e+00, 1.1000e+01,
       1.0000e+01, 8.0000e+00, 1.4000e+01, 1.5000e+01, 2.2000e+01,
       1.6000e+01, 1.3000e+01, 1.7000e+01, 1.2000e+01, 2.0000e+01,
       2.1000e+01, 2.6000e+01, 2.4000e+01, 3.6000e+01, 1.9000e+01,
       2.5000e+01, 1.8000e+01, 2.7000e+01, 2.9000e+01, 3.9000e+01,
       2.8000e+01, 3.3000e+01, 3.8000e+01, 3.7000e+01, 3.0000e+01,
       4.3000e+01, 4.2000e+01, 4.0000e+01, 3.4000e+01, 3.2000e+01,
       2.3000e+01, 3.1000e+01, 9.0000e+01, 5.5000e+01, 5.0000e+01,
       3.5000e+01, 4.5000e+01, 9.9900e+02, 5.1000e+01, 4.4000e+01,
       4.1000e+01, 4.7000e+01, 1.0400e+02, 6.5000e+01, 6.8000e+01,
       5.9000e+01, 5.4000e+01, 6.0000e+01, 8.0000e+01, 9.9999e+04,
       7.8000e+01, 6.3000e+01, 8.2000e+01, 9.5000e+01, 9.1000e+01,
       5.7000e+01, 9.7000e+01, 1.2000e+02, 7.4000e+01, 9.2000e+01,
       7.5000e+01, 4.9000e+01, 5.2000e+01, 1.1700e+02, 5.6000e

In [36]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00    NaN     NaN      Y  12/15/2020   
808126        N        N      0.00    NaN     NaN      U  12/15/2020   
808124        N        N      0.00    NaN     NaN      Y  12/15/2020   
808123        N        N      0.00    NaN     NaN      Y  12/15/2020   
808113        N        N      0.00    NaN     NaN      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [37]:
df_merge_vaers.X_STAY.unique()

array([nan, 'Y'], dtype=object)

In [38]:
df_merge_vaers['X_STAY'] = df_merge_vaers.X_STAY.fillna('N')

In [39]:
df_merge_vaers.X_STAY.unique()

array(['N', 'Y'], dtype=object)

In [40]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N     NaN      Y  12/15/2020   
808126        N        N      0.00      N     NaN      U  12/15/2020   
808124        N        N      0.00      N     NaN      Y  12/15/2020   
808123        N        N      0.00      N     NaN      Y  12/15/2020   
808113        N        N      0.00      N     NaN      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [41]:
df_merge_vaers.DISABLE.unique()

array([nan, 'Y'], dtype=object)

In [42]:
df_merge_vaers['DISABLE'] = df_merge_vaers.X_STAY.fillna('N')

In [43]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [44]:
df_merge_vaers['DISABLE'] = df_merge_vaers.X_STAY.fillna('N')

In [45]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [46]:
df_merge_vaers.RECOVD.unique()

array(['Y', 'U', nan, 'N'], dtype=object)

In [47]:
print("Unknown :- ",len(df_merge_vaers[df_merge_vaers['RECOVD'] == 'U']))
print("NaN :- ", df_merge_vaers['RECOVD'].isnull().sum())

Unknown :-  150619
NaN :-  54806


The RECOVD Column in the dataframe contains information about the status of patients recovery.
* Y -> recovered
* N -> Not recovered
* U -> Unknown (count :- 150619)
* NaN -> field is left blank (count :- 54806)

In [48]:
print("NaN values in AGE field :- ", df_merge_vaers['AGE_YRS'].isnull().sum())

NaN values in AGE field :-  58451


In the AGE_YRS columns there are many filed which are NaN (Blank) :- 58451.
* Remedy Fill NaN fileds with mean age.

In [49]:
print("NaN values in Symptoms field :- ", df_merge_vaers['SYMPTOM_TEXT'].isnull().sum())

NaN values in Symptoms field :-  126


As there are only 126 fields in symptoms column which are blank it is ok to delete such fields.

In [50]:
df_merge_vaers.dropna(subset=["SYMPTOM_TEXT"], inplace=True)
print("NaN values in Symptoms field :- ", df_merge_vaers['SYMPTOM_TEXT'].isnull().sum())
df_merge_vaers.head()

NaN values in Symptoms field :-  0


VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                          NaN   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [51]:
df_merge_vaers.LAB_DATA.unique()

array(['none', 'ED visit, blood draw, monitoring - 12/15/20', nan, ...,
       'Ct scan - negative for stroke.',
       'skin biopsy confirming bullous pemphigoid.',
       'Test Name: Blood cholesterol; Result Unstructured Data: Test Result:abnormal; Test Name: COVID-19 virus test; Test Result: Positive  ; Test Name: Vitamin D; Result Unstructured Data: Test Result:deficiency'],
      dtype=object)

In [52]:
print("NaN values in LAB Data field :- ", df_merge_vaers['LAB_DATA'].isnull().sum())


NaN values in LAB Data field :-  389034


In [53]:
df_merge_vaers['LAB_DATA'] = df_merge_vaers['LAB_DATA'].fillna('none')
print("NaN values in LAB Data field :- ", df_merge_vaers['LAB_DATA'].isnull().sum())

NaN values in LAB Data field :-  0


In [54]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                         none   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                                NaN     NaN   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none       NaN  12/15/2020   
808126  low blood sugar, low blood pressure       NaN  12/15/2020   
808124                  borderline diabetic       NaN  12/15/2020   
808123                  borderline diabetic       NaN  12/15/2020   
808113                                  NaN       NaN  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983          NaN       NaN         NaN   
808126          NaN       NaN           Y   
808124          NaN       NaN         NaN   
808123          NaN       NaN         NaN   
808113          NaN       NaN         NaN   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [55]:
df_merge_vaers.PRIOR_VAX.unique()

array([nan,
       'vaso-vagal type symptoms, similar presentation 12 hours post vaccine',
       'Tetanus shot. Near death experience. Saw tunnel, white light, passed relatives, God. Transient neurological problem lasting abou',
       ..., 'Thimerosal containing vaccines',
       'All vaccination gives her mild pains',
       'Pertussis at 1 year- 1981'], dtype=object)

In [56]:
df_merge_vaers['OTHER_MEDS'] = df_merge_vaers['OTHER_MEDS'].fillna('none')
df_merge_vaers['CUR_ILL'] = df_merge_vaers['CUR_ILL'].fillna('none')
df_merge_vaers['HISTORY'] = df_merge_vaers['HISTORY'].fillna('none')
df_merge_vaers['PRIOR_VAX'] = df_merge_vaers['PRIOR_VAX'].fillna('none')
df_merge_vaers['BIRTH_DEFECT'] = df_merge_vaers['BIRTH_DEFECT'].fillna('N')
df_merge_vaers['OFC_VISIT'] = df_merge_vaers['OFC_VISIT'].fillna('N')
df_merge_vaers['ER_ED_VISIT'] = df_merge_vaers['ER_ED_VISIT'].fillna('N')
df_merge_vaers['VAX_ROUTE'] = df_merge_vaers['VAX_ROUTE'].fillna('IM')
df_merge_vaers['ALLERGIES'] = df_merge_vaers['ALLERGIES'].fillna('none')

In [57]:
print(df_merge_vaers.shape)
df_merge_vaers.head()

(721309, 41)


VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                         none   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                               none    none   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none      none  12/15/2020   
808126  low blood sugar, low blood pressure      none  12/15/2020   
808124                  borderline diabetic      none  12/15/2020   
808123                  borderline diabetic      none  12/15/2020   
808113                                 none      none  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983            N         N           N   
808126            N         N           Y   
808124            N         N           N   
808123            N         N           N   
808113            N         N           N   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2  SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia            24.00               NaN   
808126                Pulse abnormal            23.10     Tunnel vision   
808124               Limb discomfort            23.10       Nervousness   
808123                       Fatigue            23.10  Feeling abnormal   
808113                   Feeling hot            23.10      Paraesthesia   

        SYMPTOMVERSION3              SYMPTOM4  SYMPTOMVERSION4      SYMPTOM5  \
807983              NaN                   NaN   

In [58]:
df_merge_vaers.SYMPTOM1.unique()

array(['Hypoaesthesia', 'Hyperhidrosis', 'Impaired driving ability', ...,
       'Colon neoplasm', 'Ovarian clear cell carcinoma',
       'Myelosuppression'], dtype=object)

In [59]:
print("NaN values in SYMPTOM1 field :- ", df_merge_vaers['SYMPTOM1'].isnull().sum())
print("NaN values in SYMPTOMVERSION1 field :- ", df_merge_vaers['SYMPTOMVERSION1'].isnull().sum())
print("NaN values in SYMPTOM2 field :- ", df_merge_vaers['SYMPTOM2'].isnull().sum())
print("NaN values in SYMPTOMVERSION2 field :- ", df_merge_vaers['SYMPTOMVERSION2'].isnull().sum())
print("NaN values in SYMPTOM3 field :- ", df_merge_vaers['SYMPTOM3'].isnull().sum())
print("NaN values in SYMPTOMVERSION3 field :- ", df_merge_vaers['SYMPTOMVERSION3'].isnull().sum())
print("NaN values in SYMPTOM4 field :- ", df_merge_vaers['SYMPTOM4'].isnull().sum())
print("NaN values in SYMPTOMVERSION4 field :- ", df_merge_vaers['SYMPTOMVERSION4'].isnull().sum())
print("NaN values in SYMPTOM5 field :- ", df_merge_vaers['SYMPTOM5'].isnull().sum())
print("NaN values in SYMPTOMVERSION5 field :- ", df_merge_vaers['SYMPTOMVERSION5'].isnull().sum())
print("NaN values in VAX_MANU field :- ", df_merge_vaers['VAX_MANU'].isnull().sum())
print("NaN values in VAX_ROUTE field :- ", df_merge_vaers['VAX_ROUTE'].isnull().sum())
print("NaN values in VAX_NAME field :- ", df_merge_vaers['VAX_NAME'].isnull().sum())

NaN values in SYMPTOM1 field :-  0
NaN values in SYMPTOMVERSION1 field :-  0
NaN values in SYMPTOM2 field :-  153074
NaN values in SYMPTOMVERSION2 field :-  153074
NaN values in SYMPTOM3 field :-  280201
NaN values in SYMPTOMVERSION3 field :-  280201
NaN values in SYMPTOM4 field :-  387775
NaN values in SYMPTOMVERSION4 field :-  387775
NaN values in SYMPTOM5 field :-  474692
NaN values in SYMPTOMVERSION5 field :-  474692
NaN values in VAX_MANU field :-  0
NaN values in VAX_ROUTE field :-  0
NaN values in VAX_NAME field :-  0


In [60]:
df_merge_vaers['SYMPTOM2'] = df_merge_vaers['SYMPTOM2'].fillna('none')
df_merge_vaers['SYMPTOMVERSION2'] = df_merge_vaers['SYMPTOMVERSION2'].fillna('none')
df_merge_vaers['SYMPTOM3'] = df_merge_vaers['SYMPTOM3'].fillna('none')
df_merge_vaers['SYMPTOMVERSION3'] = df_merge_vaers['SYMPTOMVERSION3'].fillna('none')
df_merge_vaers['SYMPTOM4'] = df_merge_vaers['SYMPTOM4'].fillna('none')
df_merge_vaers['SYMPTOMVERSION4'] = df_merge_vaers['SYMPTOMVERSION4'].fillna('none')
df_merge_vaers['SYMPTOM5'] = df_merge_vaers['SYMPTOM5'].fillna('none')
df_merge_vaers['SYMPTOMVERSION5'] = df_merge_vaers['SYMPTOMVERSION5'].fillna('none')

In [61]:
print("NaN values in SYMPTOM1 field :- ", df_merge_vaers['SYMPTOM1'].isnull().sum())
print("NaN values in SYMPTOMVERSION1 field :- ", df_merge_vaers['SYMPTOMVERSION1'].isnull().sum())
print("NaN values in SYMPTOM2 field :- ", df_merge_vaers['SYMPTOM2'].isnull().sum())
print("NaN values in SYMPTOMVERSION2 field :- ", df_merge_vaers['SYMPTOMVERSION2'].isnull().sum())
print("NaN values in SYMPTOM3 field :- ", df_merge_vaers['SYMPTOM3'].isnull().sum())
print("NaN values in SYMPTOMVERSION3 field :- ", df_merge_vaers['SYMPTOMVERSION3'].isnull().sum())
print("NaN values in SYMPTOM4 field :- ", df_merge_vaers['SYMPTOM4'].isnull().sum())
print("NaN values in SYMPTOMVERSION4 field :- ", df_merge_vaers['SYMPTOMVERSION4'].isnull().sum())
print("NaN values in SYMPTOM5 field :- ", df_merge_vaers['SYMPTOM5'].isnull().sum())
print("NaN values in SYMPTOMVERSION5 field :- ", df_merge_vaers['SYMPTOMVERSION5'].isnull().sum())
print("NaN values in VAX_MANU field :- ", df_merge_vaers['VAX_MANU'].isnull().sum())
print("NaN values in VAX_ROUTE field :- ", df_merge_vaers['VAX_ROUTE'].isnull().sum())
print("NaN values in VAX_NAME field :- ", df_merge_vaers['VAX_NAME'].isnull().sum())

NaN values in SYMPTOM1 field :-  0
NaN values in SYMPTOMVERSION1 field :-  0
NaN values in SYMPTOM2 field :-  0
NaN values in SYMPTOMVERSION2 field :-  0
NaN values in SYMPTOM3 field :-  0
NaN values in SYMPTOMVERSION3 field :-  0
NaN values in SYMPTOM4 field :-  0
NaN values in SYMPTOMVERSION4 field :-  0
NaN values in SYMPTOM5 field :-  0
NaN values in SYMPTOMVERSION5 field :-  0
NaN values in VAX_MANU field :-  0
NaN values in VAX_ROUTE field :-  0
NaN values in VAX_NAME field :-  0


In [62]:
df_merge_vaers.head()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
807983    902418 2020-12-15    NJ    56.00   F   
808126    902557 2020-12-15    AK    32.00   F   
808124    902524 2020-12-15    NE    59.00   F   
808123    902524 2020-12-15    NE    59.00   F   
808113    902518 2020-12-15    IN    58.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
807983  Patient experienced mild numbness traveling fr...    N       NA   
808126  1750- IM injection ot R Deltoid.  She was sitt...    N       NA   
808124  I am a immunization nurse at this location.  I...    N       NA   
808123  I am a immunization nurse at this location.  I...    N       NA   
808113  Tingling of upper lip and cheeks, warmth in fa...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD    VAX_DATE  \
807983        N        N      0.00      N       N      Y  12/15/2020   
808126        N        N      0.00      N       N      U  12/15/2020   
808124        N        N      0.00      N       N      Y  12/15/2020   
808123        N        N      0.00      N       N      Y  12/15/2020   
808113        N        N      0.00      N       N      Y  12/15/2020   

        ONSET_DATE  NUMDAYS                                     LAB_DATA  \
807983  12/15/2020     0.00                                         none   
808126  12/15/2020     0.00  ED visit, blood draw, monitoring - 12/15/20   
808124  12/15/2020     0.00                                         none   
808123  12/15/2020     0.00                                         none   
808113  12/15/2020     0.00                                         none   

                                               OTHER_MEDS CUR_ILL  \
807983                                              latex    none   
808126                                           Nuvaring    none   
808124  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808123  Dorzolamide HCL/Timolol Maleate Ophthalmic  So...    none   
808113                                               none    none   

                                    HISTORY PRIOR_VAX TODAYS_DATE  \
807983                                 none      none  12/15/2020   
808126  low blood sugar, low blood pressure      none  12/15/2020   
808124                  borderline diabetic      none  12/15/2020   
808123                  borderline diabetic      none  12/15/2020   
808113                                 none      none  12/15/2020   

       BIRTH_DEFECT OFC_VISIT ER_ED_VISIT  \
807983            N         N           N   
808126            N         N           Y   
808124            N         N           N   
808123            N         N           N   
808113            N         N           N   

                                                ALLERGIES  \
807983                                               none   
808126                                               None   
808124  Allergies: Morphine Spinal and Lumigan (Bimato...   
808123  Allergies: Morphine Spinal and Lumigan (Bimato...   
808113  Latex, Norflex, shellfish, pomegranate, almond...   

                        SYMPTOM1  SYMPTOMVERSION1  \
807983             Hypoaesthesia            24.00   
808126             Hyperhidrosis            23.10   
808124  Impaired driving ability            23.10   
808123                  Asthenia            23.10   
808113              Eye pruritus            23.10   

                            SYMPTOM2 SYMPTOMVERSION2          SYMPTOM3  \
807983  Injection site hypoaesthesia           24.00              none   
808126                Pulse abnormal           23.10     Tunnel vision   
808124               Limb discomfort           23.10       Nervousness   
808123                       Fatigue           23.10  Feeling abnormal   
808113                   Feeling hot           23.10      Paraesthesia   

       SYMPTOMVERSION3              SYMPTOM4 SYMPTOMVERSION4      SYMPTOM5  \
807983            none                  none            

In [63]:
df_merge_vaers.tail()

VAERS_ID   RECVDATE STATE  AGE_YRS SEX  \
750994   1669568 2021-09-03    MO      NaN   M   
750995   1669573 2021-09-03   NaN      NaN   U   
750996   1669577 2021-09-03   NaN      NaN   U   
750983   1669185 2021-09-03    PR    58.00   F   
750976   1667911 2021-09-03    OH    84.00   F   

                                             SYMPTOM_TEXT DIED DATEDIED  \
750994  SUSPECTED IMMUNOLOGICAL VACCINE FAILURE; This ...    N       NA   
750995  BLOOD CLOT; This spontaneous report received f...    N       NA   
750996  DEATH; This spontaneous report received from a...    Y       NA   
750983  Constant cough; Itching and swelling of the ey...    N       NA   
750976  8/16/21: COVID pos presented to ED with cough,...    N       NA   

       L_THREAT HOSPITAL  HOSPDAYS X_STAY DISABLE RECOVD   VAX_DATE  \
750994        N        N      0.00      N       N      U        NaN   
750995        N        N      0.00      N       N      U        NaN   
750996        N        N      0.00      N       N      N        NaN   
750983        N        N      0.00      N       N      N  8/17/2021   
750976        N        Y      2.00      N       N      U  2/17/2021   

       ONSET_DATE  NUMDAYS                                           LAB_DATA  \
750994        NaN      NaN  Test Name: COVID-19 antibody test; Result Unst...   
750995        NaN      NaN                                               none   
750996        NaN      NaN                                               none   
750983  8/18/2021     1.00                                               none   
750976  8/16/2021   180.00                                               none   

                                               OTHER_MEDS CUR_ILL  \
750994                                               none    none   
750995                                               none    none   
750996                                               none    none   
750983                                               none    none   
750976  acetaminophen, albuterol, alendronate, ASA, ca...    none   

                                                  HISTORY PRIOR_VAX  \
750994                                               none      none   
750995    Comments: Patient never had blood clot in past.      none   
750996                                               none      none   
750983  Medical History/Concurrent Conditions: Penicil...      none   
750976  COPD, DM, hyperlipidemia, HTN, hypothyroidism,...      none   

       TODAYS_DATE BIRTH_DEFECT OFC_VISIT ER_ED_VISIT            ALLERGIES  \
750994   8/30/2021            N         N           N                 none   
750995   8/30/2021            N         N           N                 none   
750996    9/2/2021            N         N           N                 none   
750983    9/1/2021            N         N           Y                 none   
750976    9/3/2021            N         N           N  penicillin, lipitor   

                        SYMPTOM1  SYMPTOMVERSION1               SYMPTOM2  \
750994  SARS-CoV-2 antibody test            24.00  Therapy non-responder   
750995                Thrombosis            24.00                   none   
750996                     Death            24.00                   none   
750983             Swelling face            24.00     Swelling of eyelid   
750976                  Asthenia            24.00               COVID-19   

       SYMPTOMVERSION2        SYMPTOM3 SYMPTOMVERSION3 SYMPTOM4  \
750994           24.00            none            none     none   
750995            none            none            none     none   
750996            none            none            none     none   
750983           24.00  Vision blurred           24.00     none   
750976           24.00           Cough           24.00  Fatigue   

       SYMPTOMVERSION4          SYMPTOM5 SYMPTOMVERSION5 VAX_TYPE  \
750994            none              none            none  COVID19   
750995            none           

In [64]:
print("NaN values in VAX_DATE field :- ", df_merge_vaers['VAX_DATE'].isnull().sum())
print("NaN values in ONSET_DATE field :- ", df_merge_vaers['ONSET_DATE'].isnull().sum())
print("NaN values in NUMDAYS field :- ", df_merge_vaers['NUMDAYS'].isnull().sum())
print("NaN values in AGE_YRS field :- ", df_merge_vaers['AGE_YRS'].isnull().sum())

NaN values in VAX_DATE field :-  40058
NaN values in ONSET_DATE field :-  42358
NaN values in NUMDAYS field :-  70150
NaN values in AGE_YRS field :-  58451


In [65]:
df_merge_vaers.to_csv('C:\\Users\\NIKHIL\\Desktop\\CERELABS\\final_vaers')